statsmodels/stats/contingency_tables.py

```diff
@@ -0,0 +1,1215 @@
+            hypothesis.
+        stat_sd0 : float
+            The standard deviation of the test statistic under the
+            null hypothesis.
+        zscore : float
+            The Z-score for the test statistic.
+        pvalue : float
+            The p-value for the test.
+
+        Notes
+        -----
+        Using the default row and column scores for the linear-by-linear
+        association test gives the Cochran-Armitage trend test.
+        """
+
+        if row_scores is None:
+            row_scores = np.arange(self.table.shape[0])
+
+        if col_scores is None:
+            col_scores = np.arange(self.table.shape[1])
+
+        if len(row_scores) != self.table.shape[0]:
+            raise ValueError("The length of `row_scores` must match the first dimension of `table`.")
+
+        if len(col_scores) != self.table.shape[1]:
+            raise ValueError("The length of `col_scores` must match the second dimension of `table`.")
+
+        # The test statistic
+        stat = np.dot(row_scores, np.dot(self.table, col_scores))
+
+        # Some needed quantities
+        n_obs = self.table.sum()
+        rtot = self.table.sum(1)
+        um = np.dot(row_scores, rtot)
+        u2m = np.dot(row_scores**2, rtot)
+        ctot = self.table.sum(0)
+        vn = np.dot(col_scores, ctot)
+        v2n = np.dot(col_scores**2, ctot)
+
+        # The null mean and variance of the test statistic
+        e_stat = um * vn / n_obs
+        v_stat = (u2m - um**2 / n_obs) * (v2n - vn**2 / n_obs) / (n_obs - 1)
+        sd_stat = np.sqrt(v_stat)
+
+        zscore = (stat - e_stat) / sd_stat
+        pvalue = 2 * stats.norm.cdf(-np.abs(zscore))
+
+        b = _bunch()
+        b.stat = stat
+        b.stat_e0 = e_stat
+        b.stat_sd0 = sd_stat
+        b.zscore = zscore
+        b.pvalue = pvalue
+        return b
+
+
+    @cache_readonly
+    def marginal_probabilities(self):
+        """
+        Return the estimated row and column marginal distributions.
+        """
+        n = self.table.sum()
+        row = self.table.sum(1) / n
+        col = self.table.sum(0) / n
+
+        if isinstance(self.table_orig, pd.DataFrame):
+            row = pd.Series(row, self.table_orig.index)
+            col = pd.Series(col, self.table_orig.columns)
+
+        return row, col
+
+
+    @cache_readonly
+    def independence_probabilities(self):
+        """
+        Estimated cell probabilities under independence.
+        """
+        row, col = self.marginal_probabilities
+        itab = np.outer(row, col)
+
+        if isinstance(self.table_orig, pd.DataFrame):
+            itab = pd.DataFrame(itab, self.table_orig.index,
+                                self.table_orig.columns)
+
+        return itab
+
+
+    @cache_readonly
+    def fittedvalues(self):
+        """
+        Fitted values under independence.
+        """
+        probs = self.independence_probabilities
+        fit = self.table.sum() * probs
+        return fit
```

Comment:
general question, because I looked at related questions in #2605
independence here is conditional on row and column marginal probabilities.

How does this fit together with contingency tables that 
(1) only have one margin fixed by design, e.g. comparing two independent proportions and 
(2) only have the total sample size fixed by design, e.g. two paired samples as in agreement measures.

(I still only have eclectic knowledge about contingency tables.)